backend/src/room/room.gateway.ts

```
@@ -0,0 +1,68 @@
+import { UseGuards } from "@nestjs/common";
+import {
+  ConnectedSocket,
+  MessageBody,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from "@nestjs/websockets";
+import { InjectPinoLogger, PinoLogger } from "nestjs-pino";
+import { Namespace, Socket } from "socket.io";
+
+import { WsAuthGuard } from "src/auth/ws.guard";
+
+import { RoomService } from "./room.service";
+
+import { ROOM_EVENTS, ROOM_NAMESPACE } from "~shared/constants";
+
+@UseGuards(WsAuthGuard)
+@WebSocketGateway({ namespace: ROOM_NAMESPACE })
+export class RoomGateway {
+  @WebSocketServer()
+  server: Namespace;
+
+  constructor(
+    @InjectPinoLogger()
+    private readonly logger: PinoLogger,
+    private readonly roomService: RoomService,
+  ) {}
+
+  @SubscribeMessage(ROOM_EVENTS.JOIN)
+  async handleJoin(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() roomId: string,
+  ) {
+    this.logger.info(`Joining room: ${client.id}`);
+    const userId = Number(client.handshake.headers.authorization);
+
+    if (await this.roomService.getRoom(userId)) {
```

If the `roomService` doesn't contain the room, let's throw an error instead.